clients/src/main/java/org/apache/kafka/common/metrics/stats/Histogram.java

```diff
@@ -149,10 +156,8 @@ public int toBin(double x) {
             } else if (x > this.max) {
                 return this.bins - 1;
             } else {
-                double scaled = x / this.scale;
-                return (int) (-0.5 + Math.sqrt(2.0 * scaled + 0.25));
+                return (int) (-0.5 + 0.5 * Math.sqrt(1.0 + 8.0 * x / this.scale));
```

Comment:
Not really sure where the previous equation came from, and couldn't find any similar algorithm in papers or code. The new equation is simply a simplification of the quadratic formula that solves the second-order equation on [line 149](https://github.com/apache/kafka/pull/3911/files#diff-39a49e6ddd2cffdf5f4bd40d2fd2c94fR149).

Reply:
I'm not sure what served as the original inspiration, but the algorithm is really just returning bins whose widths are the sum of all prior bin widths. IOW, the equation on line 149 in `fromBin` method is really just `n (n+1) / 2`, aka Gauss' Summation Trick, scaled such that the final bin has a range that ends with `max`.